### PR TITLE
New Resource: `azurerm_kusto_database_principal_assignment`

### PR DESCRIPTION
**Pull request by jrauschenbusch**
fixes https://github.com/terraform-providers/terraform-provider-azurerm/issues/7383

### CODE REVIEW

**Comment by jrauschenbusch** [comment]
@katbyte This is currently WIP. I'll need to check whether everything works correctly. After all we need to deprecate the old "azurerm_database_principal" resource and give a hint to the new available resource within the docs.

File: azurerm/internal/services/kusto/kusto_database_principal_assignment_resource.go [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for deletion of Kusto Database Principal Assignment %q (Resource Group %q, Cluster %q, Database %q): %+v", id.Name, id.ResourceGroup, id.Cluster, id.Database, err)
+	}
+
+	return nil
+}
+
+func validateAzureRMKustoDatabasePrincipalAssignmentName(v interface{}, k string) (warnings []string, errors []error) {
```
Comment (jrauschenbusch) [inline azurerm/internal/services/kusto/kusto_database_principal_assignment_resource.go:232]: Naming rules must be verified. Currently the same as Database naming.

**Comment by jrauschenbusch** [comment]
@mbfrahry As it's the same structure as `azurerm_kusto_cluster_principal_assignment` it could make sense that you also review this one. Notes from the other PR are already applied.

File: website/docs/r/kusto_database_principal_assignment.html.markdown [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+* `tenant_name` - The name of the tenant.
+
+## Timeouts
+
+The `timeouts` block allows you to specify [timeouts](https://www.terraform.io/docs/configuration/resources.html#timeouts) for certain actions:
+
+* `create` - (Defaults to 1 hour) Used when creating the Kusto Database Principal Assignment.
+* `read` - (Defaults to 5 minutes) Used when retrieving the Kusto Database Principal Assignment.
+* `update` - (Defaults to 1 hour) Used when updating the Kusto Database Principal Assignment.
```
Comment (jackofallops) [inline website/docs/r/kusto_database_principal_assignment.html.markdown:91]: can remove
```suggestion
```

**Comment by jrauschenbusch** [comment]
@jackofallops I've fixed everything and added the `requiresImport` acc test. And yes, everything was based on the `azurerm_kusto_cluster_principal_assignment` resource.

Please review again. I had to do a `go mod vendor` which updated the vendors folder. I don't know if the diffs should really be there. Please consider it within the review.

File: vendor/github.com/hashicorp/go-azure-helpers/authentication/config.go [hunk 3]
```diff
@@ -14,6 +14,7 @@ import (
 // new Azure management client.
 type Config struct {
 	ClientID           string
+	ClientSecret       string
```
Comment (jrauschenbusch) [inline vendor/github.com/hashicorp/go-azure-helpers/authentication/config.go:17]: This diff is definitively not intended to be here.

Comment (jackofallops) [reply vendor/github.com/hashicorp/go-azure-helpers/authentication/config.go:17]: Thanks @jrauschenbusch - I'll fetch your branch and figure out what's going on, we had to make an emergency fix to the master branch on Monday which is likely to be the root of the issue.

Comment (jackofallops) [reply vendor/github.com/hashicorp/go-azure-helpers/authentication/config.go]: I _think_ it's sorted now, will let Travis do it's thing and loop back shortly, we're a bit stretched today.

**Comment by jackofallops** [comment]
Tests passing:
image

**Pull request merged**